### ISSUE TITLE
HaikuWebKit: Build failed with "error: Building release without compiler optimizations"

### ISSUE THREAD

**Issue by waddlesplash** [haikuports/haikuports#6364]
https://build.haiku-os.org/buildmaster/master/x86_64/?buildrunDir=6299&viewMode=expanded
```
In file included from /sources/haikuwebkit-HaikuWebKit-1.8.3/Source/WTF/wtf/Platform.h:31,
                from /sources/haikuwebkit-HaikuWebKit-1.8.3/Source/WTF/wtf/ExportMacros.h:32,
                from /sources/haikuwebkit-HaikuWebKit-1.8.3/Source/JavaScriptCore/runtime/JSExportMacros.h:32,
                from /sources/haikuwebkit-HaikuWebKit-1.8.3/Source/JavaScriptCore/config.h:33,
                from /sources/haikuwebkit-HaikuWebKit-1.8.3/Source/JavaScriptCore/disassembler/udis86/udis86.c:27:
/sources/haikuwebkit-HaikuWebKit-1.8.3/Source/WTF/wtf/Compiler.h:103:2: error: #error "Building release without compiler optimizations: WebKit will be slow. Set -DRELEASE_WITHOUT_OPTIMIZATIONS if this is intended."
 103 | #error "Building release without compiler optimizations: WebKit will be slow. Set -DRELEASE_WITHOUT_OPTIMIZATIONS if this is intended."
     |  ^~~~~
```

**Comment by waddlesplash** [comment]
I see we are passing `-DCMAKE_BUILD_TYPE=Release` to CMake, though. So I wonder what's gone wrong here?

**Comment by waddlesplash** [comment]
Indeed it appears the full command line that appears in the log has no `-O`...

**Comment by waddlesplash** [comment]
The GCC2 build has not failed, as of yet: https://build.haiku-os.org/buildmaster/master/x86_gcc2/?buildrunDir=2026&viewMode=expanded

**Comment by pulkomandy** [comment]
CMAKE_BUILD_TYPE does not seem to enable or disable optimizations by default, and I see nothing in haikuwebkit code that would do it either. So it looks like we are expected to enable it ourselves? But it isn't done in other platform CMakeLists either.

**Comment by waddlesplash** [comment]
CMAKE_BUILD_TYPE=Release is definitely supposed to enable optimizations: https://github.com/Kitware/CMake/blob/8f67d42671942aebfe345a574c563f59efe931d0/Modules/Compiler/GNU.cmake#L58

**Comment by pulkomandy** [comment]
This looks suspicious:

    -- The C compiler identification is TinyCC

TinyCC does not seem to be in the list of activated packages. What happened here?

**Comment by pulkomandy** [comment]
To debug this I would need the CMakeOutput.log and CMakeError.log files from a failed build. They are in WebKitBuild/Release/CMakeFiles/

Can anyone share these? @Vidrep did you get this problem locally? That would be easier than getting the files from the buildbots.

**Comment by pulkomandy** [comment]
Fixed with 6a7f34b76acc482a9f84cbe9d9ca8ae20c05d08a.
